kyo-prelude/shared/src/main/scala/kyo2/kernel/Effect.scala

```diff
@@ -159,26 +159,24 @@ object Effect:
             handle3Loop(v, Context.empty)
         end apply
 
-        inline def state[I[_], O[_], E <: Effect[I, O], State, A, B, S, S2](
+        inline def state[I[_], O[_], E <: Effect[I, O], State, A, B, S, S2, S3](
             inline tag: Tag[E],
             inline state: State,
             v: A < (E & S)
         )(
-            inline handle: Safepoint ?=> [C] => (I[C], State, O[C] => A < (E & S & S2)) => (State, A < (E & S & S2)),
-            inline done: (State, A) => B < (S & S2) = (_: State, v: A) => v,
+            inline handle: Safepoint ?=> [C] => (I[C], State, O[C] => A < (E & S & S2)) => (State, A < (E & S & S2)) < S3,
```

Comment:
This is something the previous core used to support and was missing from the new kernel. It allows `handle` to perform other effects before returning the tuple, which is something necessary for `Stream` since methods like `take` have to emit values of the same type that is getting handled.